pkg/sink/webhook.go

```diff
@@ -73,49 +50,24 @@ func NewWebhook(log logrus.FieldLogger, c config.Webhook, reporter AnalyticsRepo
 	return whNotifier, nil
 }
 
-// SendEvent sends event notification to Webhook url
-func (w *Webhook) SendEvent(ctx context.Context, event event.Event, eventSources []string) (err error) {
-	if !sliceutil.Intersect(w.Bindings.Sources, eventSources) {
-		w.log.Debugf("Event sources do not match Webhook sources, event: %+v, eventSources: %+v", event, eventSources)
-		return nil
-	}
+// SendMessageToAll is no-op.
+func (w *Webhook) SendMessageToAll(_ context.Context, _ interactive.CoreMessage) error {
+	return nil
+}
 
+// SendMessage is no-op.
+func (w *Webhook) SendMessage(ctx context.Context, _ interactive.CoreMessage, sources []string, rawData any) error {
 	jsonPayload := &WebhookPayload{
-		EventMeta: EventMeta{
-			Kind:      event.Kind,
-			Name:      event.Name,
-			Namespace: event.Namespace,
-			Cluster:   event.Cluster,
-		},
-		EventStatus: EventStatus{
-			Type:     event.Type,
-			Level:    event.Level,
-			Reason:   event.Reason,
-			Error:    event.Error,
-			Messages: event.Messages,
-		},
-		EventSummary:    format.ShortMessage(event),
-		TimeStamp:       event.TimeStamp,
-		Recommendations: event.Recommendations,
-		Warnings:        event.Warnings,
+		Source: sources[0],
+		Data:   rawData,
```

Comment:
Actually we cannot keep the previous format since, that format was all about kubernetes events. I applied same strategy we did for telemetry. So, those information is called RawData and we send it to webhook and es. Notice that, those data is coming from plugins which can be anything

We haven't changed the signature for sources yet, it is still accepting list of sources, and that is what I get only first since there is only one source inside it. this will be removed soon, and we will have only single source.

Reply:
This is good suggestion :)